contracts/stellar-multicall/src/types.rs

```diff
@@ -0,0 +1,9 @@
+use soroban_sdk::{contracttype, Address, Symbol, Val, Vec};
+
+#[contracttype]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct FunctionCall {
+    pub contract: Address,
```

Comment:
```suggestion
    pub approver: Address,
    pub contract: Address,
```
for each function call, `approver.require_auth()` should be called so that an attacker can't frontrun the individual function calls. `approver` can be set to the caller, even if the function doesn't need an approval